### PR TITLE
[WIP] plot_age_pyramid: add the possibility to have one datetime_ref that differs for each patient

### PR DESCRIPTION
**Pull request by strayMat**
The previous plot_age_pyramid function only allow for one fixed datetime reference for all persons. 

In some case studies , we would like to have a different datetime of reference for each person. 
Eg. the date of their inclusion in the study. The objective is to draw the pyramid of ages  of patient at inclusion. (Very useful and more pertinent when we have cohorts over a long time period). 

## Description

Add to plot_age_pyramid the possibility to give the datetime_ref argument as a column name. 

## Checklist

- [x] Changes were documented in the changelog (pending section).
- [x] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

File: tests/test_age_pyramid.py [hunk 1]
```diff
@@ -8,31 +8,36 @@
 
 from eds_scikit.datasets.synthetic.person import load_person
 from eds_scikit.plot.data_quality import plot_age_pyramid
+import pandas as pd 
+import numpy as np
 
 data = load_person()
 
+person_with_inclusion_date = data.person.copy()
+person_with_inclusion_date["inclusion_datetime"] = person_with_inclusion_date["birth_datetime"] + pd.to_timedelta(np.random.randint(0, 1000, len(person_with_inclusion_date)), unit='d')
```
Comment (Vincent-Maladiere) [inline tests/test_age_pyramid.py:17]: Nice idea. There's a lot going in this line, could you add some extra steps for readability?
Like so:
```python
N = len(person_with_inclusion_date)
delta_days = pd.to_timedelta(np.random.randint(0, 1000, N), unit='d')
person_with_inclusion_date["inclusion_datetime"] = (
    person_with_inclusion_date["birth_datetime"]
    + delta_days
)
```
Also, let's apply Black formater.

File: tests/test_age_pyramid.py [hunk 2]
```diff
@@ -8,31 +8,36 @@
...
 
-def test_plot_age_pyramid():
-    original_person = data.person.copy()
-
-    datetime_ref = datetime(2020, 1, 1)
-    chart, group_gender_age = plot_age_pyramid(data.person, datetime_ref, savefig=False)
+@pytest.mark.parametrize(
```
Comment (Vincent-Maladiere) [inline tests/test_age_pyramid.py:19]: Could you split this test into two functions?
1. The first one (this function) asserts that `datetime_ref` is correct. You should also add a parseable date string in the mark input.
2. The second one checks that errors are returned from incorrect `datetime_ref` (all error cases should be tested to maximize test coverage) by leveraging `pytest.raises`.

**Review by Vincent-Maladiere — CHANGES_REQUESTED** [review]
Hey, @strayMat thank you for this PR! Below are some comments before merging.

File: eds_scikit/plot/data_quality.py [hunk 3]
```diff
@@ -55,11 +57,15 @@ def plot_age_pyramid(
...
+        datetime_ref_ = pd.to_datetime(datetime_ref)
+    elif type(datetime_ref) == str:
+        if datetime_ref not in person_.columns:
+            raise ValueError(f"{datetime_ref} should be either a column of the dataframe, or a datetime.")
+        datetime_ref_ = person_[datetime_ref]
     else:
-        today = datetime.today()
-    person_["age"] = (today - person_["birth_datetime"]).dt.total_seconds()
+        datetime_ref_ = datetime.today()
+    person_["age"] = (datetime_ref_ - person_["birth_datetime"]).dt.total_seconds()
```
[the remark this answers is not in the document]
Comment (strayMat) [reply eds_scikit/plot/data_quality.py]: ok

File: eds_scikit/plot/data_quality.py [hunk 4]
```diff
@@ -55,11 +57,15 @@ def plot_age_pyramid(
...
 
-    if datetime_ref:
-        today = pd.to_datetime(datetime_ref)
+    if type(datetime_ref) == datetime:
+        datetime_ref_ = pd.to_datetime(datetime_ref)
+    elif type(datetime_ref) == str:
+        if datetime_ref not in person_.columns:
+            raise ValueError(f"{datetime_ref} should be either a column of the dataframe, or a datetime.")
+        datetime_ref_ = person_[datetime_ref]
     else:
```
[the remark this answers is not in the document]
Comment (strayMat) [reply eds_scikit/plot/data_quality.py:76]: thx ! looks much better now.

File: eds_scikit/plot/data_quality.py [hunk 5]
```diff
@@ -55,11 +57,15 @@ def plot_age_pyramid(
 
     person_ = person.copy()
 
-    if datetime_ref:
-        today = pd.to_datetime(datetime_ref)
+    if type(datetime_ref) == datetime:
+        datetime_ref_ = pd.to_datetime(datetime_ref)
+    elif type(datetime_ref) == str:
```
[the remark this answers is not in the document]
Comment (strayMat) [reply eds_scikit/plot/data_quality.py]: got it!

File: eds_scikit/plot/data_quality.py [hunk 6]
```diff
@@ -55,11 +57,15 @@ def plot_age_pyramid(
 
     person_ = person.copy()
 
-    if datetime_ref:
-        today = pd.to_datetime(datetime_ref)
+    if type(datetime_ref) == datetime:
```
[the remark this answers is not in the document]
Comment (strayMat) [reply eds_scikit/plot/data_quality.py]: got it!

File: tests/test_age_pyramid.py (hunk 2 above)
Comment (strayMat) [reply tests/test_age_pyramid.py:25]: new test added ! Thanks for the review :heart:

File: changelog.md [hunk 7]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v0.1.3 (2022-12-06)
+
+- Adding person-dependant `datetime_ref` to `plot_age_pyramid`
+
```
Comment (Thomzoy) [inline changelog.md:6]:
```suggestion
## Pending

### Added

- Adding person-dependant `datetime_ref` to `plot_age_pyramid`

```

**Comment by Vincent-Maladiere** [comment]
Looks good! We need to merge #20 before we merge this PR as well.

**Comment by strayMat** [comment]
Where are we for this PR ? I am not sure to see what is missing before merging ?

**Comment by Vincent-Maladiere** [comment]
Close in favor of #43 to avoid nested merging conflicts.

**Pull request closed without merging**